### ISSUE TITLE
Clear cache on pages where images have changed

### ISSUE THREAD

**Issue by lenvanessen** [putyourlightson/craft-blitz#431]
We have a problem with regards to changing the images focal point. If you change the focal point of an image, in Craft 4.0.5.2, the image transforms are immediatly deleted. But since the page has already been cached, the page is full of 404's where it already contains the final url to the transformed image, while it really should have contained the image generate transform url.

Anyway, like with other element references, Blitz could track this and invalidate the caches?

**Comment by bencroker** [comment]
Makes sense, I'll have to look into the broader implications of doing this though. Will report back when I have an update.

**Comment by bencroker** [comment]
I forgot that changing the focal point doesn't trigger an asset save, see https://github.com/putyourlightson/craft-blitz/issues/235#issuecomment-655592940.

**Comment by lenvanessen** [comment]
@bencroker I believe it does to be honest. Because as a workaround for this problem, we've implemented the following, and it works like a charm (other than that it clears everything, not just the affected pages)
```
 // untill https://github.com/putyourlightson/craft-blitz/issues/431
        Event::on(
            Asset::class,
            Asset::EVENT_BEFORE_SAVE,
            function(ModelEvent $event) {
                Blitz::$plugin->clearCache->clearAll();
            }
        );
        ```

**Comment by bencroker** [comment]
Ok so it may have changed since then, I'll look into it again in more detail.

**Comment by bencroker** [comment]
Managed to track down the issue and have fixed in https://github.com/putyourlightson/craft-blitz/commit/8494164b5afe13f6ca98ea6d8632dc29c12dc111 for the next release.

**Comment by bencroker** [comment]
Released in version 4.2.0.
